### PR TITLE
[PART-1] TW-1690: Implement search external channel

### PR DESCRIPTION
**Pull request by hieutbui**
## Ticket
- #1690 

## DOD:
- [x] API
- [x] Datasource
- [x] Repository
- [x] Interactor

### CODE REVIEW

File: lib/data/model/search/public_room_request.dart [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+part 'public_room_request.g.dart';
+
+@JsonSerializable()
+class PublicRoomRequest with EquatableMixin {
+  @JsonKey(name: 'filter')
+  final PublicRoomQueryFilter? filter;
+
+  @JsonKey(name: 'include_all_networks')
+  final bool? includeAllNetworks;
```
Comment (hoangdat) [inline lib/data/model/search/public_room_request.dart:13]: remove

File: lib/data/model/search/public_room_request.dart [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  @JsonKey(name: 'limit')
+  final int? limit;
+
+  @JsonKey(name: 'since')
+  final String? since;
+
+  @JsonKey(name: 'third_party_instance_id')
+  final String? thirdPartyInstanceId;
```
Comment (hoangdat) [inline lib/data/model/search/public_room_request.dart:22]: remove

File: lib/di/global/get_it_initializer.dart [hunk 3]
```diff
@@ -350,6 +367,8 @@ class GetItInitializer {
     getIt.registerSingleton<DownloadMediaFileInteractor>(
       DownloadMediaFileInteractor(),
     );
+
+    getIt.registerSingleton<PublicRoomInteractor>(PublicRoomInteractor());
```
Comment (hoangdat) [inline lib/di/global/get_it_initializer.dart:371]:
```suggestion
    getIt.registerFactory<PublicRoomInteractor>(PublicRoomInteractor());
```

File: lib/data/model/search/public_room_request.dart (hunk 1 above)
Comment (hieutbui) [reply lib/data/model/search/public_room_request.dart]: done

File: lib/data/model/search/public_room_request.dart (hunk 2 above)
Comment (hieutbui) [reply lib/data/model/search/public_room_request.dart]: done

File: lib/di/global/get_it_initializer.dart (hunk 3 above)
Comment (hieutbui) [reply lib/di/global/get_it_initializer.dart]: done

**Review by Te-Z — APPROVED** [review]

**Review by sherlockvn — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**